### PR TITLE
Rework coordEach implementation to match setup from geomEach

### PR DESCRIPTION
**Pull request by tobrun**
This PR proposes the refactored code from #50 to match geomEach setup. 

@baparham @lukas-h 
- wdyt? 
- is this better as the original ported turf-js code? 
- Would you take a different approach?

### CODE REVIEW

**Comment by tobrun** [comment]
After doing an additional improvement for geometry collection, I'm getting similar results as old implementation

### This branch

```
coordEach
  ✓ geometry (0 us)
  ✓ feature (0 us)
  ✓ geometry collection (0 us)
  ✓ feature collection (220 us)
  ✓ ./test/examples/rfc7946/featureCollectionWithBBox1.geojson (0 us)
  ✓ ./test/examples/rfc7946/polygon2.geojson (1 us)
  ✓ ./test/examples/rfc7946/antimeridianCutting2.geojson (1 us)
  ✓ ./test/examples/rfc7946/point.geojson (0 us)
  ✓ ./test/examples/rfc7946/featureWithBBox.geojson (0 us)
  ✓ ./test/examples/rfc7946/antimeridianCutting1.geojson (0 us)
  ✓ ./test/examples/rfc7946/lineString.geojson (0 us)
  ✓ ./test/examples/rfc7946/featureCollection.geojson (1 us)
  ✓ ./test/examples/rfc7946/polygon1.geojson (0 us)
  ✓ ./test/examples/rfc7946/geometryCollection.geojson (0 us)
  ✓ ./test/examples/rfc7946/featureCollectionWithBBox2.geojson (0 us)
  ✓ ./test/examples/rfc7946/multiPolygon.geojson (2 us)
  ✓ ./test/examples/rfc7946/multiPoint.geojson (0 us)
  ✓ ./test/examples/rfc7946/multiLineString.geojson (1 us)
  ✓ ./test/examples/wikipedia/polygon2.geojson (1 us)
  ✓ ./test/examples/wikipedia/point.geojson (0 us)
  ✓ ./test/examples/wikipedia/lineString.geojson (0 us)
  ✓ ./test/examples/wikipedia/multiPolygon1.geojson (1 us)
  ✓ ./test/examples/wikipedia/featureCollection.geojson (1 us)
  ✓ ./test/examples/wikipedia/polygon1.geojson (0 us)
  ✓ ./test/examples/wikipedia/geometryCollection.geojson (0 us)
  ✓ ./test/examples/wikipedia/multiPolygon2.geojson (2 us)
  ✓ ./test/examples/wikipedia/multiPoint.geojson (0 us)
  ✓ ./test/examples/wikipedia/multiLineString.geojson (1 us)
  ✓ ./test/examples/lukas-h/germany-states.geojson (2 ms)
  ✓ ./test/examples/bbox/polygon.geojson (0 us)
  ✓ ./test/examples/bbox/point.geojson (0 us)
  ✓ ./test/examples/bbox/lineString.geojson (0 us)
  ✓ ./test/examples/bbox/geometryCollection.geojson (0 us)
  ✓ ./test/examples/bbox/multiPolygon.geojson (1 us)
  ✓ ./test/examples/bbox/multiPoint.geojson (0 us)
  ✓ ./test/examples/bbox/multiLineString.geojson (0 us)

```

### Old branch

```
 coordEach
  ✓ geometry (0 us)
  ✓ feature (0 us)
  ✓ geometry collection (0 us)
  ✓ feature collection (245 us)
  ✓ ./test/examples/rfc7946/featureCollectionWithBBox1.geojson (0 us)
  ✓ ./test/examples/rfc7946/polygon2.geojson (1 us)
  ✓ ./test/examples/rfc7946/antimeridianCutting2.geojson (1 us)
  ✓ ./test/examples/rfc7946/point.geojson (0 us)
  ✓ ./test/examples/rfc7946/featureWithBBox.geojson (0 us)
  ✓ ./test/examples/rfc7946/antimeridianCutting1.geojson (1 us)
  ✓ ./test/examples/rfc7946/lineString.geojson (0 us)
  ✓ ./test/examples/rfc7946/featureCollection.geojson (1 us)
  ✓ ./test/examples/rfc7946/polygon1.geojson (1 us)
  ✓ ./test/examples/rfc7946/geometryCollection.geojson (0 us)
  ✓ ./test/examples/rfc7946/featureCollectionWithBBox2.geojson (0 us)
  ✓ ./test/examples/rfc7946/multiPolygon.geojson (2 us)
  ✓ ./test/examples/rfc7946/multiPoint.geojson (0 us)
  ✓ ./test/examples/rfc7946/multiLineString.geojson (1 us)
  ✓ ./test/examples/wikipedia/polygon2.geojson (1 us)
  ✓ ./test/examples/wikipedia/point.geojson (0 us)
  ✓ ./test/examples/wikipedia/lineString.geojson (0 us)
  ✓ ./test/examples/wikipedia/multiPolygon1.geojson (1 us)
  ✓ ./test/examples/wikipedia/featureCollection.geojson (1 us)
  ✓ ./test/examples/wikipedia/polygon1.geojson (1 us)
  ✓ ./test/examples/wikipedia/geometryCollection.geojson (0 us)
  ✓ ./test/examples/wikipedia/multiPolygon2.geojson (2 us)
  ✓ ./test/examples/wikipedia/multiPoint.geojson (0 us)
  ✓ ./test/examples/wikipedia/multiLineString.geojson (1 us)
  ✓ ./test/examples/lukas-h/germany-states.geojson (2 ms)
  ✓ ./test/examples/bbox/polygon.geojson (1 us)
  ✓ ./test/examples/bbox/point.geojson (0 us)
  ✓ ./test/examples/bbox/lineString.geojson (0 us)
  ✓ ./test/examples/bbox/geometryCollection.geojson (0 us)
  ✓ ./test/examples/bbox/multiPolygon.geojson (1 us)
  ✓ ./test/examples/bbox/multiPoint.geojson (0 us)
  ✓ ./test/examples/bbox/multiLineString.geojson (1 us)

**Comment by lukas-h** [comment]
> * Would you take a different approach?

I have built an implementation where I could get rid of many of the helper functions by implementing it internally with `geomEach`. I just wanted to test if that is possible, but seems to work. You can find it here: https://github.com/dartclub/turf_dart/blob/c5f6a05b9700a4572d14729d8cde022d3e3264ea/lib/src/meta.dart#L19

File: lib/src/meta.dart [hunk 1]
```diff
@@ -18,132 +18,182 @@ typedef CoordEachCallback = dynamic Function(
...
-      geometryMaybeCollection = isFeatureCollection
-          ? geoJSON.features[featureIndex].geometry
-          : isFeature
-              ? geoJSON.geometry
-              : geoJSON;
-
-      isGeometryCollection = geometryMaybeCollection != null
-          ? geometryMaybeCollection is GeometryCollection
-          : false;
+    if (geoJSON is GeometryCollection) {
```
[the remark this answers is not in the document]
Comment (tobrun) [reply lib/src/meta.dart:23]: that is definitely a mistake

**Comment by tobrun** [comment]
> I've merged https://github.com/dartclub/turf_dart/pull/63 – I'll close this PR, right? @tobrun

But you merged it into the branch of this PR, your version isn't available on main yet :smile: 
Feel free to create a new PR or reopen this one, up to you

**Comment by lukas-h** [comment]
Oh haha,

**Pull request closed without merging**